src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -940,7 +940,10 @@ def _discoverAMI(self) -> str:
         :return: The AMI ID (a string like 'ami-0a9a5d2b65cce04eb') for Flatcar.
         :rtype: str
         """
-        return get_flatcar_ami(self.aws.client(self._region, 'ec2'))
+        try:
+            return get_flatcar_ami(self.aws.client(self._region, 'ec2'), self._architecture)
+        except:
```

Comment:
This probably wants to be `except AttributeError` and have a comment about how sometimes we are running in the in-workflow Toil scaler and self._architecture isn't set.

Also, if we don't pass an architecture, the default is amd64, which only happens to work because the in-workflow scaler is only ever used with amd64 Mesos clusters. We might need to note that or this will seem wrong.

Reply:
This might be better as a getattr call with a default value.